src/bin/list_store.rs

```diff
@@ -0,0 +1,299 @@
+        },
+        Data {
+            fixed: false,
+            number: 50055,
+            severity: "Normal".to_string(),
+            description: "Rework width/height computations for TreeView".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 58278,
+            severity: "Normal".to_string(),
+            description: "gtk_dialog_set_response_sensitive () doesn't work".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 55767,
+            severity: "Normal".to_string(),
+            description: "Getters for all setters".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 56925,
+            severity: "Normal".to_string(),
+            description: "Gtkcalender size".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 56221,
+            severity: "Normal".to_string(),
+            description: "Selectable label needs right-click copy menu".to_string(),
+        },
+        Data {
+            fixed: true,
+            number: 50939,
+            severity: "Normal".to_string(),
+            description: "Add shift clicking to GtkTextView".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 6112,
+            severity: "Normal".to_string(),
+            description: "netscape-like collapsable toolbars".to_string(),
+        },
+        Data {
+            fixed: false,
+            number: 1,
+            severity: "Normal".to_string(),
+            description: "First bug :=)".to_string(),
+        },
+    ];
+
+    let store = gtk::ListStore::new(&col_types);
+
+    let col_indices: [u32; 8] = [0, 1, 2, 3, 4, 5, 6, 7];
+
+    for (d_idx, d) in data.iter().enumerate() {
+        let icon_name = if d_idx == 1 || d_idx == 3 {
+            "battery-caution-charging-symbolic"
+        } else {
+            ""
+        };
+
+        let sensitive = d_idx != 3;
+
+        let values: [&dyn ToValue; 8] = [
+            &d.fixed.to_value(),
+            &d.number.to_value(),
+            &d.severity.to_value(),
+            &d.description.to_value(),
+            &0u32.to_value(),
+            &icon_name.to_value(),
+            &false.to_value(),
+            &sensitive.to_value(),
+        ];
+        store.set(&store.append(), &col_indices, &values);
+    }
+
+    store
+}
+
+fn fixed_toggled<W: IsA<CellRendererToggle>>(model: &gtk::ListStore, _w: &W, path: TreePath) {
+    let iter = model.get_iter(&path).unwrap();
+    let mut fixed = model.get_value(&iter, COLUMN_FIXED).get::<bool>().unwrap();
+    fixed = !fixed;
+    model.set_value(&iter, COLUMN_FIXED as u32, &fixed.to_value());
+}
+
+fn add_columns(model: Rc<gtk::ListStore>, treeview: &TreeView) {
+    // Column for fixed toggles
+    {
+        let renderer = CellRendererToggle::new();
+        renderer.connect_toggled(move |w, path| fixed_toggled(&*model, w, path));
```

Comment:
This is so much nicer with `ListBox` and simply being able to bind a property of the model with the property on the widget :) Oh well!

Reply:
Can you elaborate? This is porting a C example without much changes, but I'm interested in learning other ways of doing this.

Reply:
That was just a comment about the uglyness of the `TreeView` / `ListStore` API, nothing wrong with your code.

Take a look at the `listbox_model.rs` example for something similar with `ListBox`. There the connection between the model and the UI is a simple property binding.